examples/headsail-bsp/src/lib.rs

```diff
@@ -43,11 +43,17 @@ mod sysctrl;
 #[cfg(feature = "panic-uart")]
 mod ufmt_panic;
 
+#[cfg(feature = "hpc")]
+const EXTERNAL_BIT: usize = 0x1_0000_0000;
+
 /// # Safety
 ///
 /// Unaligned reads may fail to produce expected results on RISC-V.
 #[inline(always)]
 pub unsafe fn read_u8(addr: usize) -> u8 {
+    #[cfg(feature = "hpc")]
+    return core::ptr::read_volatile((addr | EXTERNAL_BIT) as *const _);
+    #[cfg(not(feature = "hpc"))]
     core::ptr::read_volatile(addr as *const _)
```

Comment:
Inconsistent return statement & return expression. Use return expression in both.

Reply:
These changes were removed compleatly